cores/esp8266/Esp.cpp

```diff
@@ -107,11 +107,19 @@ void EspClass::wdtFeed(void)
 
 extern "C" void esp_yield();
 
-void EspClass::deepSleep(uint64_t time_us, WakeMode mode)
+bool EspClass::deepSleep(uint64_t time_us, WakeMode mode)
 {
+    if (time_us > deepSleepMax())
+    {
+       #ifdef DEBUG_SERIAL
+          DEBUG_SERIAL.println("Error: max sleeptime exceeded");
+       #endif
+       return 0; // error: max sleeptime exceeded
```

Comment:
return type is bool, but returned value is int => return false here

Reply:
You're right.
Strange though that this is not detected by git code check.